document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/SmallFullAccountsDataHandler.java

```diff
@@ -0,0 +1,158 @@
+package uk.gov.companieshouse.document.generator.accounts.handler.accounts;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.model.accounts.CompanyAccounts;
+import uk.gov.companieshouse.document.generator.accounts.AccountType;
+import uk.gov.companieshouse.document.generator.accounts.LinkType;
+import uk.gov.companieshouse.document.generator.accounts.data.transaction.Transaction;
+import uk.gov.companieshouse.document.generator.accounts.exception.HandlerException;
+import uk.gov.companieshouse.document.generator.accounts.exception.ServiceException;
+import uk.gov.companieshouse.document.generator.accounts.mapping.smallfull.model.ixbrl.SmallFullAccountIxbrl;
+import uk.gov.companieshouse.document.generator.accounts.service.AccountsService;
+import uk.gov.companieshouse.document.generator.accounts.service.CompanyService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.document.generator.accounts.AccountsDocumentInfoServiceImpl.MODULE_NAME_SPACE;
+
+@Component
+public class SmallFullAccountsDataHandler {
```

Comment:
Looking to the future, will we have a class for each type of accounts? If so, in the `getDocumentInfo` method we could use a factory to return the correct type of `AccountsDataHandler` for a provided set of accounts. That would make this service much more easily expandable for further account types, like Micros and Dormants.